File: AppTool/src/main/java/com/luoye/apptool/viewbinding/BaseViewFragment.java
package com.luoye.apptool.viewbinding;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.DefaultLifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.viewbinding.ViewBinding;

import com.luoye.apptool.utils.ViewBindingUtil;

public abstract class BaseViewFragment<T extends ViewBinding> extends Fragment implements DefaultLifecycleObserver {

    protected String TAG = "---BaseViewFragment";
    protected Context context;
    protected Activity activity;
    protected T binding;


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        context = getActivity();
        activity = getActivity();
        getLifecycle().addObserver(this);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = ViewBindingUtil.create(getClass(), inflater, container);
        initFragment();
        return binding.getRoot();
    }

    protected abstract void initFragment();

    @Override
    public void onStart(@NonNull LifecycleOwner owner) {
        DefaultLifecycleObserver.super.onStart(owner);
    }
}

File: ktBase/src/main/java/com/luoye/ktbase/binding/viewbinding/fragment/BaseViewFragment.kt
package com.luoye.ktbase.binding.viewbinding.fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewbinding.ViewBinding
import com.luoye.ktbase.binding.ViewBindingUtil

/**
 *
 * Created by: luoye
 * Time: 2022/11/30
 * user:
 *
 */
abstract class BaseViewFragment<V : ViewBinding> : Fragment() {

    protected var TAG = "---BaseViewFragment"
    protected lateinit var binding: V


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        avedInstanceState: Bundle?
    ): View {
        binding = ViewBindingUtil.create(javaClass, inflater, container)
        initFragment()
        return binding.getRoot()
    }

    protected abstract fun initFragment()

}